### PR TITLE
Filtering the webserver information in the nginx default error responses

### PR DESCRIPTION
**Pull request by jashaik**
Signed-off-by: jan shahid shaik <jashaik@progress.com>

### Description

The internal server software used by the chef-server is revealed by the web server(nginx/openresty) in the default error reponses.

### Issues Resolved

https://app.zenhub.com/workspaces/chef-infra-server-team-5fc64867d45ca500173dbbc7/issues/chef/customer-bugs/373

**Solved -** 

- openresty version/openresty server info in all the success/opscode-erchef generated responses - Solved by 'server_tokens off;' nginx tag
- openresty verison info in all the default nginx error responses - Solved by 'server_tokens off;' nginx tag
- openresty server name in the default nginx error reponses - solved by the 'body_filter_by_lua' directive(PR #2539 ). 

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by jashaik** [comment]
Pipeline Information -
Adhoc - https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/2790#be532555-d0af-47a9-b7a8-3f99ffdfc1a5
Umbrella - https://buildkite.com/chef/chef-umbrella-master-chef-server/builds/501#d7422619-0db3-46b7-906d-60e5b3f4dd40

File: omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb [hunk 1]
```diff
@@ -148,6 +150,13 @@ http {
             more_clear_headers Server;
             access_log /var/log/opscode/nginx/rewrite-port-80.log;
             return 301 https://$host$request_uri;
+            location @errorrespfilter {
+              return 301 https://$host$request_uri;
+              header_filter_by_lua_block { ngx.header.content_length = nil }
+              body_filter_by_lua '
+              ngx.arg[1] = ngx.re.sub(ngx.arg[1],"openresty", "chef")
```
Comment (PrajaktaPurohit) [inline omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb:157]: Should this be changed to ngx.arg[1] = ngx.re.sub(ngx.arg[1],"openresty", "") or some non consequential string instead?

**Comment by PrajaktaPurohit** [comment]
current output
```
root@api:~# curl -i -X TRACE -k https://api.chef-server.dev/_status
HTTP/1.1 301 Moved Permanently
Date: Mon, 21 Jun 2021 19:00:53 GMT
Content-Type: text/html
Transfer-Encoding: chunked
Connection: close
Location: https://api.chef-server.dev/_status

<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>chef</center>
</body>
</html>
```

**Review by PrajaktaPurohit — APPROVED** [review]
Nicely done! Left some suggestions. https://github.com/chef/chef-server/blob/master/oc-chef-pedant/spec/api/header_spec.rb might be a good place to add some tests for this.

File: omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb (hunk 1 above)
Comment (marcparadise) [reply omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb:157]: Agree, if the idea is that we don't want to reveal anything about the system from the error.

File: omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb [hunk 2]
```diff
@@ -148,6 +150,13 @@ http {
             more_clear_headers Server;
             access_log /var/log/opscode/nginx/rewrite-port-80.log;
             return 301 https://$host$request_uri;
+            location @errorrespfilter {
```
Comment (marcparadise) [inline omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx.conf.erb:153]: Let's include a comment here that explains why we're doing this filter.  Same in api lb conf.

**Comment by jashaik** [comment]
> Nicely done! Left some suggestions. https://github.com/chef/chef-server/blob/master/oc-chef-pedant/spec/api/header_spec.rb might be a good place to add some tests for this.

This issue is happening when we use curl, So not able to add tests for this.Please find the below screenshot for reference
Screenshot 2021-06-22 at 12 09 33 PM

**Review by marcparadise — APPROVED** [review]

**Pull request merged**